projects/schematics/src/migrations/4_0/constructor-deprecations/data/dynamic-attribute.service.migration.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  DYNAMIC_ATTRIBUTE_SERVICE,
+  SMART_EDIT_SERVICE,
+  SPARTACUS_CORE,
+  UNIFIED_INJECTOR,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const DYNAMIC_ATTRIBUTE_SERVICE_MIGRATION: ConstructorDeprecation = {
+  // projects/core/src/cms/services/dynamic-attribute.service.ts
+  class: DYNAMIC_ATTRIBUTE_SERVICE,
+  importPath: SPARTACUS_CORE,
+  deprecatedParams: [
+    {
+      className: SMART_EDIT_SERVICE,
```

Comment:
👍 

was confused at first, but then I found it was only smart_edit_service that was always available before from https://github.com/SAP/spartacus/pull/10700/files